File: Data Structures & Algorithms/subsets/submission-1.java
class Solution {
    public void backtrack(int[] nums,List<Integer> curr,int pos, List<List<Integer>> res){
        if(pos==nums.length){
            res.add(new ArrayList<>(curr));
            return;
        }
        curr.add(nums[pos]);
        backtrack(nums,curr,pos+1,res);
        curr.remove(curr.size()-1);
        backtrack(nums,curr,pos+1,res);
    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res=new ArrayList<>();
        backtrack(nums,new ArrayList<>(), 0,res);
        return res;
        
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        int l=0,r=0;
        int n=nums.size();
        vector<int> res;
        int currwinmax=INT_MIN;
        int currwinmaxidx=-1;
        while(r<n){
            int curr=nums[r];
            if(currwinmax<curr){
                currwinmax=curr;
            currwinmaxidx=r;
            }
            r++;
            if(r-l==k){
                res.push_back(currwinmax);
                if(l==currwinmaxidx){
                    currwinmax=INT_MIN;
                    currwinmaxidx=-1;
                    for(int i=l+1;i<r;i++){
                        if(nums[i]>=currwinmax){
                            currwinmax=nums[i];
                            currwinmaxidx=i;
                        }
                    }
                }
                l+=1;
            }

        }
        return res;
    }
};
